### ISSUE TITLE
some queries

### ISSUE THREAD

**Issue by soans1994** [robertklee/KITTI-RoadSeg#6]
hello author,

great work with the keras tensorflow implementation. i have some doubts regarding your implementation.
1. why is the DEFAULT_EPOCHS = 20 and DEFAULT_BATCH_SIZE = 12. is it because of small dataset?
2. image size = 640x192, did you match this with the image generator function?
3. how can i implement a simple encoder decoder using vgg 16.

thank you

**Comment by robertklee** [comment]
Thanks for reaching out. See replies inline:

1. why is the DEFAULT_EPOCHS = 20 and DEFAULT_BATCH_SIZE = 12. is it because of small dataset?

These were just parameters I had set at the beginning that I forgot to update. Batch size is limited by GPU memory, and Epochs is limited by size of dataset (which is fairly small here).

2. image size = 640x192, did you match this with the image generator function?

Yes, this was matched with the image generator function. Note in `constants.py` that:
``` Python
input_shape = (640,192,3)
input_shape_full_size = (1024,320,3)
```

I resize the input image down to 640x192.

3. how can i implement a simple encoder decoder using vgg 16.
    
You can make the changes in `model.py`'s function:
``` Python
def create_Model(input_shape=constants.input_shape, encoder_type=constants.EncoderType.resnet50):
```

**Comment by soans1994** [comment]
thank you for your reply, 

1. how did you process the road_data dataset? i am not able to understand the segmentation generate function. did you use the the segmentation color images directly? can you explain me the final data used for training?
2. i used the vgg-16 encoder and unet decoder model for achieving segmentation from 
https://github.com/nikhilroxtomar/Semantic-Segmentation-Architecture/blob/main/TensorFlow/vgg16_unet.py

edit: i fixed the error. changed the last conv layer (2,1). but i had to change the import headers from tensorflow.keras to keras. to match your train code. but when i try his other segmentation models, i get error because of some layers cant be found in keras. package. 

from keras.layers import Conv2D, BatchNormalization, Activation, MaxPool2D, Conv2DTranspose, Concatenate, Input
from keras.models import Model
from keras.applications import VGG16

3. i want to ask how to plot the loss curve from the log directory.
thank you

**Comment by robertklee** [comment]
Hey sorry I didn't get notified about your reply. 

1. I used the segmentation map colours. There's actually a small bug where there was one additional colour I didn't account for so you might want to fix it. On some of the images the oncoming traffic lane has a different colour. 

2. Glad to know you figured it out. :)

3. I used Tensorboard to plot the loss curves. It's built in and you can see the code in `train.py`

**Comment by soans1994** [comment]
thank you for your reply.

the kitti dataset has keypoints annotation file and also the color segmentation images. you didn't use the calib folder(which contains the keypoints annoataions) right?

**Comment by robertklee** [comment]
The calibration files aren't required for this approach. They are used to transform from the provided prospective images into birds eye view space.

**Comment by robertklee** [comment]
Also, if there is an issue with my code, feel free to put up a pull request! :)

I am happy to accept contributions.

**Comment by soans1994** [comment]
Thank you very much👍🏼
